H/HELICS/build_tarballs.jl

```diff
@@ -15,15 +15,28 @@
 
 using BinaryBuilder
 
-HELICS_VERSION = v"2.8.0"
-HELICS_SHA = "f2b218494407573c75561b7d4d656bc60f7592e970dd87d98c969066d76d89c1"
+HELICS_VERSION = v"3.1.1"
+HELICS_SHA = "65791ddede5f06aff58daa7cf0b9d244baa6c4e08e119ca010ddcc6137a479d9"
 
 sources = [
     ArchiveSource("https://github.com/GMLC-TDC/HELICS/releases/download/v$HELICS_VERSION/Helics-v$HELICS_VERSION-source.tar.gz",
                   "$HELICS_SHA"),
+    ArchiveSource("https://github.com/phracker/MacOSX-SDKs/releases/download/10.15/MacOSX10.15.sdk.tar.xz",
+                  "2408d07df7f324d3beea818585a6d990ba99587c218a3969f924dfcc4de93b62"),
 ]
 
 script = raw"""
+if [[ "${target}" == x86_64-apple-darwin* ]]; then
+    export MACOSX_DEPLOYMENT_TARGET=10.15
+    export CXXFLAGS="-mmacosx-version-min=10.15"
+    export CFLAGS="-mmacosx-version-min=10.15"
+    pushd $WORKSPACE/srcdir/MacOSX10.*.sdk
+    rm -rf /opt/${target}/${target}/sys-root/System
+    cp -ra usr/* "/opt/${target}/${target}/sys-root/usr/."
+    cp -ra System "/opt/${target}/${target}/sys-root/."
+    popd
+fi
```

Comment:
This wouldn't be necessary with https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/193, in the meantime we have to do it manually:
```suggestion
elif [[ "${target}" == aarch64-apple-darwin* ]]; then
    # While waiting for https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/193 to be merged
    export CXXFLAGS="-mmacosx-version-min=11.0"
fi
```